### PR TITLE
Re-enable some ref tests

### PR DESCRIPTION
**Pull request by yannham**
Depends on #72.

Enable part of the test suite on references. For now, we've simply re-enabled the tests that are passing with the current state of the repository, which at least means those won't regress in a future update. ~~It would be better to see, for each failing test, if it's failing because it's unsupported or because there's a bug in the current implementation -  this is left for a future PR~~ (I made another pass and commented why each test doesn't pass in the current state of js2n)

### CODE REVIEW

File: tests/json_schema_test_suite_test.rs [hunk 1]
```diff
@@ -23,7 +28,33 @@ use stringreader::StringReader;
...
+    "ref_27_1.*", // urn:uuid URI scheme not supported,
+    "ref_28_0.*", // external reference (remote URI)
+    "ref_29_0.*", // external reference (remote URI)
+    "ref_2_1.*", // internal ref to an array element (#/items/0)
+    "ref_30_0.*", // external reference (remote URI)
+    "ref_31_1.*", // external reference (absolute path /absref/foobar.json)
+    "ref_34_1.*", // non top-level definition ("#/definitions//definitions/")
+    "ref_3_5.*", // schemars doesn't properly percent-decode URIs
+    "ref_5_1.*", // not related to external ref, but js2n doesn't properly ignore other components
+                 // when the `ref` field is used
```
Comment (vkleen) [inline tests/json_schema_test_suite_test.rs:55]: I guess this one should be fixed soon, as it's an actual bug in json-schema-to-nickel, not just a missing feature.

**Review by vkleen — APPROVED** [review]
Thanks a lot for documenting the reasons for the failing testcases! I don't remember why we disabled the `definitions_.*` tests, could that have been because of references?

File: tests/json_schema_test_suite_test.rs (hunk 1 above)
Comment (yannham) [reply tests/json_schema_test_suite_test.rs:55]: Yes. I wanted to proceed with this PR anyway, but I agree. I just currently don't know the JSON schema spec enough to be sure of the exact rule to follow, but I just notice that the example schema has both `$ref` set and a array schema that says that length must be at most 2, the ref is just a general array schema, and the example has 3 elements and must be passing. So it really feels like what's tested is that the initial array schema must be ignored.

Comment (vkleen) [reply tests/json_schema_test_suite_test.rs:55]: Then let's proceed the way it is, and figure out the JSON schema spec later :+1:

**Comment by yannham** [comment]
>  I don't remember why we disabled the definitions_.* tests, could that have been because of references?

I didn't see those. I re-enabled them. They consist of two tests, one passing, and one failing because of a remote URI. I've thus only banned the failing one. That leaves one new passing test, which is something :stuck_out_tongue:

**Pull request merged**